bftengine/src/preprocessor/messages/PreProcessBatchRequestMsg.cpp

```diff
@@ -52,6 +52,36 @@ void PreProcessBatchRequestMsg::validate(const ReplicasInfo& repInfo) const {
     LOG_ERROR(logger(), "Message sender is invalid" << KVLOG(senderId()));
     throw std::runtime_error(__PRETTY_FUNCTION__);
   }
+  if (!checkElements()) {
+    LOG_ERROR(logger(), "One or more PreProcessReqMsg in the list is invalid");
+    throw std::runtime_error(__PRETTY_FUNCTION__);
+  }
+}
+bool PreProcessBatchRequestMsg::checkElements() const {
```

Comment:
Is there any common code you detect? If yes, maybe we should consider putting another common class between
PreProcessBatchRequestMsg/ClientBatchRequestMsg and MessageBase? 
These 2 functions are almost identical?

Same comments like in ClientBatchRequestMsg::checkElements()

Reply:
Not sure I understood. do you want to discuss that on zoom?

Reply:
yes, lets keep it that way... it will just complicate things.